programs/sys-util/actions/chain.cpp

```diff
@@ -1,107 +1,1082 @@
+      }
+
+      // MERGE OBJECTS
+      fc::mutable_variant_object override_v{fc::json::from_file(genesis_override_file)};
+      result_v = base_v(override_v);
+    }
+    auto now = std::chrono::system_clock::now();
+    auto epoch_ms = std::chrono::time_point_cast<std::chrono::milliseconds>(now);
+    auto timestamp = std::format("{:%FT%T}", epoch_ms);
+    auto chain_id_data = std::format("wire-{}", timestamp);
+    auto chain_id = fc::sha256::hash(chain_id_data).str();
+    std::println(std::cout, "chain_id_data: {}, chain_id: {}", chain_id_data, chain_id);
+    result_v("initial_timestamp", timestamp);
+    result_v("initial_key", sysio_account_detail.public_key_str);
+    result_v("initial_chain_id", chain_id);
+
+    // Write resulting JSON
+    std::println(std::cout, "Writing genesis JSON to '{}'", target_genesis_file.generic_string());
+    fc::json::save_to_file(result_v, target_genesis_file, true);
+    std::println(std::cout, "Saved genesis JSON to '{}'", target_genesis_file.generic_string());
+
+    // DISPOSE OF THE WALLET MANAGER & EXPLICITLY STOP KIOD
+    wallet_manager->lock(wallet_default_name);
+    wallet_manager.reset();
+    stop_kiod();
+
+    if (!can_write_ini) {
+      std::println(
+        std::cerr,
+        "Refusing to overwrite existing file without --overwrite: {}",
+        target_ini_file.generic_string()
+      );
+    } else {
+      auto default_ini_file = template_dir / "config.template.ini";
+
+      if (!std::filesystem::exists(default_ini_file)) {
+        std::println(
+          std::cerr,
+          "Default config.ini file not found for template '{}': {}",
+          template_name,
+          default_ini_file.generic_string()
+        );
+        return -1;
+      }
+      std::println(
+        std::cout,
+        "Copying default_ini_file({}) to target_ini_file({})",
+        default_ini_file.generic_string(),
+        target_ini_file.generic_string()
+      );
+      if (std::filesystem::exists(ini_override_file)) std::println(
+        std::cerr,
+        "INI OVERRIDE IS NOT IMPLEMENTED YET: {}",
+        ini_override_file.generic_string()
+      );
+      std::filesystem::copy_file(default_ini_file, target_ini_file, std::filesystem::copy_options::overwrite_existing);
+
+      std::println(std::cout, "Saved config.ini to '{}'", target_ini_file.generic_string());
+    }
+
+
+    // CONFIG IS COMPLETE, NOW POPULATE THE CHAIN
+    std::println(std::cout, "Starting the chain and importing contracts");
+    std::atomic_int app_thread_status{-1};
+    std::mutex app_thread_mutex;
+    std::condition_variable app_thread_cond;
+    if (!application::null_app_singleton()) {
+      application::instance().quit();
+      application::reset_app_singleton();
+    }
+    appbase::scoped_app app;
+
+    auto on_app_init = [&](std::int32_t code) {
+      std::scoped_lock lock(app_thread_mutex);
+      app_thread_status = code;
+      app_thread_cond.notify_all();
+    };
+
+    std::thread app_thread(
```

Comment:
This seems rather odd to me. Why have sys-util duplicate the main code of nodeop. Why not just launch nodeop as a process?